connection_notifier.hh

```diff
@@ -20,19 +20,46 @@
  */
 #pragma once
 
-#include "gms/inet_address.hh"
+#include "db/query_context.hh"
+
+#include <seastar/net/inet_address.hh>
 #include <seastar/core/sstring.hh>
+#include "seastarx.hh"
+
 #include <optional>
 
+namespace db::system_keyspace {
+extern const char *const CLIENTS;
+}
+
 enum class client_type {
     cql = 0,
     thrift,
     alternator,
 };
 
+sstring to_string(client_type ct);
+
+enum class changed_column {
+    username = 0,
+    connection_stage,
+    driver_name,
+    driver_version,
+    hostname,
+    protocol_version,
+};
+
+template <changed_column column> constexpr const char* column_literal = "";
+template <> constexpr const char* column_literal<changed_column::username> = "username";
+template <> constexpr const char* column_literal<changed_column::connection_stage> = "connection_stage";
+template <> constexpr const char* column_literal<changed_column::driver_name> = "driver_name";
+template <> constexpr const char* column_literal<changed_column::driver_version> = "driver_version";
+template <> constexpr const char* column_literal<changed_column::hostname> = "hostname";
+template <> constexpr const char* column_literal<changed_column::protocol_version> = "protocol_version";
+
 // Representation of a row in `system.clients'. std::optionals are for nullable cells.
```

Comment:
Maybe add another template for the type? So if the caller passes the wrong type, we don't get a runtime error.

Reply:
Do you mean templating `const char*`?

Reply:
@avikivity ?

Reply:
I see all the types are text, so not much point.

Reply:
Actually, we also have inet_address. I thought to have two template types for column_literal, one for the ID, one for the type, so if you use the wrong type you get a compile time error. But it's not worth the effort.